Expensify/App
# Allow editing of Workspace names

### Details
Adds an edit button at the right side of the workspace name that opens a modal where you can change the workspace name.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/171962

### Tests/QA Steps
1. Create or open a workspace
2. Click the shown in along with the avatar
3. A modal with an avatar editor and a name input should open. 
4. In that modal, change everything and press "Save"
5. The modal should close, and everything you changed in the modal should get updated everywhere.

### Tested On


### Screenshots


#### Web




https://user-images.githubusercontent.com/6759612/127823948-a77f5fc0-4565-4892-a774-1169c49b2aa1.mov




#### Mobile Web




https://user-images.githubusercontent.com/6759612/127824562-43a49165-7921-430a-ae07-5e24f59e9bda.mov




#### Desktop



https://user-images.githubusercontent.com/6759612/127823948-a77f5fc0-4565-4892-a774-1169c49b2aa1.mov



#### iOS




https://user-images.githubusercontent.com/6759612/127823077-9a68475e-4371-417c-8333-ba92ff8d3a27.mov




#### Android




https://user-images.githubusercontent.com/6759612/127823730-836191ae-aceb-4b45-83d4-975cf1da20b2.mov
